src/main/java/io/github/arrayv/sortdata/SortInfo.java

```diff
@@ -123,6 +186,35 @@ public SortInfo(Sort sort) {
         this(-1, sort);
     }
 
+    private static String findSortCategory(Class<? extends Sort> sortClass) {
+        Package checkPackage = sortClass.getPackage();
+        if (checkPackage != null) {
+            SortPackageMeta packageMetaAnnotation = checkPackage.getAnnotation(SortPackageMeta.class);
+            if (packageMetaAnnotation != null) {
+                return packageMetaAnnotation.category();
+            }
+        }
+        throw new NullPointerException(
+            "Sort " + sortClass.getSimpleName() + " does not declare a category, and neither do any of its packages"
+        );
+    }
+
+    private static String requireName(String name) {
+        return Objects.requireNonNull(name, NAME_MUST_BE_SPECIFIED);
+    }
+
+    private static String normalizeName(SortMeta meta) {
+        String name = meta.name();
+        if (name.endsWith(" Sort")) {
+            return name.substring(0, name.length() - 5);
+        }
+        if (name.endsWith("sort")) {
+            return name.substring(0, name.length() - 4);
+        }
```

Comment:
I want this to be as robust as possible, so I'd prefer to keep the checks.

Reply:
then tell sort-writers rather than letting them keep bad habit (and without the checks, they'd notice the names are wrong)

(also please don't resolve unless we both agree)

Reply:
Good point. It should add to the warnings list.

Reply:
true, that's a nice idea